## Importing Libraries and Dataset

In [1]:
import numpy as np
import pandas as pd

In [2]:
import os
os.chdir('D:/Pro/02-Artificial-Intelligence-Course/01-DataScienceWithPython/02-Projects/Project-walmart_store_sales')

In [3]:
pwd

'D:\\Pro\\02-Artificial-Intelligence-Course\\01-DataScienceWithPython\\02-Projects\\Project-walmart_store_sales'

In [4]:
salesData = pd.read_csv('Walmart_Store_sales.csv')

----

## Data Understanding

In [5]:
salesData.shape

(6435, 8)

In [6]:
salesData.columns

Index(['Store', 'Date', 'Weekly_Sales', 'Holiday_Flag', 'Temperature',
       'Fuel_Price', 'CPI', 'Unemployment'],
      dtype='object')

In [7]:
salesData.head()

Store        Date  Weekly_Sales  Holiday_Flag  Temperature  Fuel_Price  \
0      1  05-02-2010    1643690.90             0        42.31       2.572   
1      1  12-02-2010    1641957.44             1        38.51       2.548   
2      1  19-02-2010    1611968.17             0        39.93       2.514   
3      1  26-02-2010    1409727.59             0        46.63       2.561   
4      1  05-03-2010    1554806.68             0        46.50       2.625   

          CPI  Unemployment  
0  211.096358         8.106  
1  211.242170         8.106  
2  211.289143         8.106  
3  211.319643         8.106  
4  211.350143         8.106

In [8]:
salesData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6435 entries, 0 to 6434
Data columns (total 8 columns):
Store           6435 non-null int64
Date            6435 non-null object
Weekly_Sales    6435 non-null float64
Holiday_Flag    6435 non-null int64
Temperature     6435 non-null float64
Fuel_Price      6435 non-null float64
CPI             6435 non-null float64
Unemployment    6435 non-null float64
dtypes: float64(5), int64(2), object(1)
memory usage: 402.3+ KB


In [9]:
salesData.describe()

Store  Weekly_Sales  Holiday_Flag  Temperature   Fuel_Price  \
count  6435.000000  6.435000e+03   6435.000000  6435.000000  6435.000000   
mean     23.000000  1.046965e+06      0.069930    60.663782     3.358607   
std      12.988182  5.643666e+05      0.255049    18.444933     0.459020   
min       1.000000  2.099862e+05      0.000000    -2.060000     2.472000   
25%      12.000000  5.533501e+05      0.000000    47.460000     2.933000   
50%      23.000000  9.607460e+05      0.000000    62.670000     3.445000   
75%      34.000000  1.420159e+06      0.000000    74.940000     3.735000   
max      45.000000  3.818686e+06      1.000000   100.140000     4.468000   

               CPI  Unemployment  
count  6435.000000   6435.000000  
mean    171.578394      7.999151  
std      39.356712      1.875885  
min     126.064000      3.879000  
25%     131.735000      6.891000  
50%     182.616521      7.874000  
75%     212.743293      8.622000  
max     227.232807     14.313000

In [10]:
salesData.isna().sum()

Store           0
Date            0
Weekly_Sales    0
Holiday_Flag    0
Temperature     0
Fuel_Price      0
CPI             0
Unemployment    0
dtype: int64

---

## Data Understanding Report

There are 6435 rows and 8 columns.

There is only one object type feature - Date.

There is no null values in data provided.

---

## Data Exploration

In [11]:
stores=salesData['Store'].unique()

print(len(stores))

45


There are 45 unique stores

## Analysis Tasks - Basic Statistics tasks

# Which store has maximum sales

In [12]:
salesData.groupby(['Store'])

In [13]:
salesData.sort_values(['Store','Weekly_Sales'],ascending=False).groupby('Store')

In [14]:
salesData[['Store','Date','Weekly_Sales']].sort_values(['Weekly_Sales'],ascending=False).groupby('Store').head(1).head(5)

Store        Date  Weekly_Sales
1905     14  24-12-2010    3818686.45
2763     20  24-12-2010    3766687.43
1333     10  24-12-2010    3749057.69
527       4  23-12-2011    3676388.98
1762     13  24-12-2010    3595903.20

The Store - 14 had made the maximum sales on 24-12-2010

-----

## Which store has maximum standard deviation i.e., the sales vary a lot. Also, find out the coefficient of mean to standard deviation

In [15]:
salesData[['Weekly_Sales']].groupby(salesData['Store']).std().sort_values('Weekly_Sales',ascending=False).head(3)

Weekly_Sales
Store               
14     317569.949476
10     302262.062504
20     275900.562742

The Store 14 has the Maximum Standard Deviation of 317569.949476

Calculate Cofficient of Mean to Standard Deviation

----